backend/controllers/message_controller.js

```diff
@@ -13,29 +13,22 @@ const Message = require("../models/Message.js");
 // 7. If saving succeeds, call `addMessageToChat` to associate the message with
 //    the chat.
 const create_message = async (request, response) => {
-  try {
+  
     const chat_id = extract_chat_id(request);
     const user_id = extract_user_id(request);
     const message = extract_message(request);
 
     const input_status = await validate_inputs(chat_id, user_id, message);
     if (input_status.status == 400) {
-      respond_with_error_json(response, input_status);
+      // respond_with_error_json(response, input_status);
+      throw new Error(`${input_status}`);
       return;
     }
-
+    
     const message_id = await save_message(message, user_id);
+    
     await add_message_to_chat(message_id, chat_id);
-    response.status(201).json({
-      status: 201,
-      message: `${message}`,
-    });
-  } catch (error) {
-    respond_with_error_json(response, {
-      status: 400,
-      error: `${error}`,
-    });
-  }
```

Comment:
I think you put the responses back. No point in removing them except for making it hard for the client to get the status.

Reply:
if i respond in here, i won't be able to respond in my code and it throws an error that can't send response after it closes